src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -3720,6 +3720,18 @@ internal uint GetValEscape(BoundExpression expr, uint scopeOfTheContainingExpres
 
                     return escape;
 
+                case BoundKind.DynamicObjectCreationExpression:
+                    var dynamicObjectCreation = (BoundDynamicObjectCreationExpression)expr;
+
+                    escape = GetValEscape(dynamicObjectCreation.Arguments, scopeOfTheContainingExpression);
+
+                    if (dynamicObjectCreation.InitializerExpressionOpt != null)
+                    {
+                        escape = Math.Max(escape, GetValEscape(dynamicObjectCreation.InitializerExpressionOpt, scopeOfTheContainingExpression));
+                    }
+
+                    return escape;
```

Comment:
>return escape; [](http://example.com/codeflow?start=20&length=14)

If dynamic object creation expression is always returnable, then implementation could be much simpler. Otherwise, we should have tests that demonstrate that the implementation makes an observable difference, i.e. is able to detect a violation in otherwise valid code.